### ISSUE TITLE
[Enhancement]: GitHub Action - Build Check on PR

### ISSUE THREAD

**Issue by maelgangloff** [PapillonApp/papillon-v7#177]
### Description du problème

Dans votre template de PR, on peut lire :
>  Vous avez testé de build le projet avec vos modifications et ce build a réussi

Pour un relecteur, il est difficile de s'assurer que le build s'effectue sans erreur.
En effet, télécharger la branche distante, construire le build et l'exécuter dans un environnement de "test" demande beaucoup de temps. Cette procédure n'est pas systématiquement effectuée pour des améliorations mineures.

### Description de l'amélioration

Il pourrait être intéressant de créer un workflow GitHub Action pour vérifier si le build de l'application est possible avant la fusion dans le dépôt principal.

Cela permettra de tester les erreurs de syntaxe et de code style. 

### Contexte supplémentaire

Si vous ne souhaitez pas exécuter ce workflow à chaque PR, il est possible que son déclenchement soit manuel.

**Comment by Gabriel29306** [comment]
Je suis d'accord avec le fait de le déclencher manuellement, seulement une fois la PR validée par tout les personnes concernées. Il ne sert à rien de tester un build si la PR n'est pas prête. 

Un autre avantage est d'éviter une attaque via des workflow malveillants dans les PRs.

**Comment by maelgangloff** [comment]
> Un autre avantage est d'éviter une attaque via des workflow malveillants dans les PRs.

Ce n'est pas le workflow du fork qui est exécuté mais celui du dépôt principal. Pas de risque à ce niveau.

**Comment by tom-theret** [comment]
Puis si tout le monde valide mais que le Build ne fonctionne pas alors, il ne sera pas possible de merge la PR.

**Comment by Louis454545** [comment]
> Puis si tout le monde valide mais que le Build ne fonctionne pas alors, il ne sera pas possible de merge la PR.

Je pense qu'on peut close l'issue

**Comment by Louis454545** [comment]
Car @yannouuuu ne veut pas de ca

**Comment by camarm-dev** [comment]
Je déterre l'issue, j'ai ajouté un workflow lint / typecheck (#264), il faut toujours ajouter un workflow de build vous pensez ? Si c'est le cas je peux essayer de m'en occuper.

**Comment by Kgeek33** [comment]
perso, je pense qu'il faut refaire plusieurs choses : 

- Ajouter un workflow pour tester les builds peut être utile
- Par rapport à ton workflow, il est très bien et super utile mais le problème, c'est que quand qqun ajoute un nv service ou tout simplement une pr, personne ne respecte le typage, chiant un peu

J'pense qu'il faut refaire plusieurs changements au niveau des pr, notamment le temps d'attente -_- ...

**Comment by camarm-dev** [comment]
> perso, je pense qu'il faut refaire plusieurs choses :
> 
>     * Ajouter un workflow pour tester les builds peut être utile
> 
>     * Par rapport à ton workflow, il est très bien et super utile mais le problème, c'est que quand qqun ajoute un nv service ou tout simplement une pr, personne ne respecte le typage, chiant un peu
> 
> 
> J'pense qu'il faut refaire plusieurs changements au niveau des pr, notamment le temps d'attente -_- ...

C'est pas compliqué de respecter le typage après ça s'appelle coder en **typescript**... Le workflow force un peu les choses il faudrait à minima qu'il passe sur la branche main vraiment... Je comprend que si il passe pas sur main, les gens vont pas corriger les erreurs dans leurs PR mais juste typer son propre code c'est pas énorme quoi.

Il faut un peu de collaboration de la part des mainteneurs, qui forcent le merge (de leurs propres PR surtout) alors que le workflow ne passe pas.

Sinon je vais check pour un workflow de build ducoup.

**Comment by camarm-dev** [comment]
Le workflow de build c'est peut être too much en vrai ? Surtout que ça va être un peu embêtant à mettre en place... Au final ça apporte rien parceque le projet peut très bien build mais quand même contenir des erreurs, je pense que c'et un peu inutile, il doit forcément y avoir du test humain. Je peux quand même essayer, mais le plus efficace serait du test unitaire de chaque module et des tests e2e (mais là bon courage pour tester toute l'app 😭)

**Comment by Kgeek33** [comment]
> Il faut un peu de collaboration de la part des mainteneurs, qui forcent le merge (de leurs propres PR surtout) alors que le workflow ne passe pas.

Je suis d'accord avec toi

**Comment by Kgeek33** [comment]
> Le workflow de build c'est peut être too much en vrai ? Surtout que ça va être un peu embêtant à mettre en place... Au final ça apporte rien parceque le projet peut très bien build mais quand même contenir des erreurs, je pense que c'et un peu inutile, il doit forcément y avoir du test humain. Je peux quand même essayer, mais le plus efficace serait du test unitaire de chaque module et des tests e2e (mais là bon courage pour tester toute l'app 😭)

- Utile ? Oui
- Facile à mettre en place ? Absolument pas 🤣

En fait, j'pense que la mieux reviendrait à faire comme la PR #260 (_abandonné on dirait_), comme ça on sait déjà que si le QR Code s'affiche, c'est que la build a réussi, et plus simple à tester

**Comment by camarm-dev** [comment]
> > Le workflow de build c'est peut être too much en vrai ? Surtout que ça va être un peu embêtant à mettre en place... Au final ça apporte rien parceque le projet peut très bien build mais quand même contenir des erreurs, je pense que c'et un peu inutile, il doit forcément y avoir du test humain. Je peux quand même essayer, mais le plus efficace serait du test unitaire de chaque module et des tests e2e (mais là bon courage pour tester toute l'app 😭)
> 
>     * Utile ? Oui
> 
>     * Facile à mettre en place ? Absolument pas 🤣
> 
> 
> En fait, j'pense que la mieux reviendrait à faire comme la PR #260 (_abandonné on dirait_), comme ça on sait déjà que si le QR Code s'affiche, c'est que la build a réussi, et plus simple à tester

Ouais c'est un bon compromis 👍🏻

**Comment by Louis454545** [comment]
> > Le workflow de build c'est peut être too much en vrai ? Surtout que ça va être un peu embêtant à mettre en place... Au final ça apporte rien parceque le projet peut très bien build mais quand même contenir des erreurs, je pense que c'et un peu inutile, il doit forcément y avoir du test humain. Je peux quand même essayer, mais le plus efficace serait du test unitaire de chaque module et des tests e2e (mais là bon courage pour tester toute l'app 😭)
> 
> - Utile ? Oui
> - Facile à mettre en place ? Absolument pas 🤣
> 
> En fait, j'pense que la mieux reviendrait à faire comme la PR #260 (_abandonné on dirait_), comme ça on sait déjà que si le QR Code s'affiche, c'est que la build a réussi, et plus simple à tester

Non, le QR code peut quand même s'afficher s'il y a une erreur

**Comment by Kgeek33** [comment]
Ah ouais ?
Logiquement ça devrait pas non ?

**Comment by camarm-dev** [comment]
> Ah ouais ? Logiquement ça devrait pas non ?

Bah ça revient à ce que je disais enft ça peut build mais quand même contenir des erreurs... Pas de syntaxe ni code dcp mais des erreurs de fonctionnements des services...

**Comment by Kgeek33** [comment]
ah ben là, on est de nouveau bloqué...

**Comment by camarm-dev** [comment]
> ah ben là, on est de nouveau bloqué...

Nan faut juste que Rémy finisse son idée et ce ser parfait ! En gros ça assure que l'app "build" et ça rend le test humain super facile !

**Comment by Kgeek33** [comment]
d'accord d'accord, alors je vais apporter de nouveaux éléments (j'avais pas vu que ça avait bougé mdr)
